### PR TITLE
Renaming Git Provider Token header

### PR DESCRIPTION
**Pull request by luizbafilho**
Closes: #971 

**What changed?**
- A different header should be used instead in order to distinguish between API endpoint authorization (i.e. listing clusters or templates) and endpoints that require interaction with GH or other git providers

**Why?**

**How did you test it?**

**Release notes**

**Documentation Changes**

### CODE REVIEW

**Review by J-Thompson12 — APPROVED** [review]
LGTM

File: pkg/server/middleware/middleware.go [hunk 1]
```diff
@@ -74,14 +74,17 @@ type contextVals struct {
 
 type key int
 
-const tokenKey key = iota
-const GRPCAuthMetadataKey = "grpc-auth"
+const (
+	tokenKey               key = iota
+	GRPCAuthMetadataKey        = "grpc-auth"
+	GitProviderTokenHeader     = "X-Git-Provider-Token"
```
Comment (bigkevmcd) [inline pkg/server/middleware/middleware.go:80]: https://datatracker.ietf.org/doc/html/rfc6648

File: ui/contexts/AppContext.tsx [hunk 2]
```diff
@@ -52,7 +52,7 @@ function wrapClient<T>(client: any, tokenGetter: () => string): T {
         ...options,
         headers: new Headers({
           ...(options.headers || {}),
-          Authorization: `token ${tokenGetter()}`,
+          "X-Git-Provider-Token": `token ${tokenGetter()}`,
```
Comment (jpellizzari) [inline ui/contexts/AppContext.tsx:55]: Can we make this a `const` as well? For computed keys in JS objects:
```suggestion
          [gitProviderTokenHeader]: `token ${tokenGetter()}`,
```

**Review by jpellizzari — APPROVED** [review]

**Comment by bigkevmcd** [comment]
This is for this PR https://github.com/weaveworks/weave-gitops/pull/965/files#diff-142329b4654445c09cf14ebc4afba7677e064ede74cb549a81e935b414156dd2R248 and the header in there is `Git-Provider-Token`.

File: pkg/server/middleware/middleware.go (hunk 1 above)
Comment (luizbafilho) [reply pkg/server/middleware/middleware.go:80]: sounds good, I'll remove the `X-` prefix

**Pull request merged**